File: NyayaSetu-backend/models/Case.js
const mongoose = require('mongoose');

const caseSchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, 'Case title is required'],
    trim: true
  },
  description: {
    type: String,
    required: [true, 'Case description is required']
  },
  client: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  lawyer: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  category: {
    type: String,
    required: true,
    enum: ['employment', 'family', 'criminal', 'corporate', 'real_estate', 'intellectual_property', 'other']
  },
  status: {
    type: String,
    enum: ['pending', 'active', 'in_progress', 'completed', 'closed'],
    default: 'pending'
  },
  priority: {
    type: String,
    enum: ['low', 'medium', 'high', 'urgent'],
    default: 'medium'
  },
  progress: {
    type: Number,
    min: 0,
    max: 100,
    default: 0
  },
  budget: {
    type: Number,
    default: 0
  },
  hoursLogged: {
    type: Number,
    default: 0
  },
  deadline: {
    type: Date
  },
  documents: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Document'
  }],
  timeline: [{
    event: String,
    description: String,
    date: {
      type: Date,
      default: Date.now
    },
    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User'
    }
  }],
  messages: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Message'
  }]
}, {
  timestamps: true
});

module.exports = mongoose.model('Case', caseSchema);

File: app/src/main/java/com/nyayasetu/app/models/Case.kt
package com.nyayasetu.app.models

data class Case(
    val id: String = "",
    val title: String = "",
    val description: String = "",
    val clientId: String = "",
    val clientName: String = "",
    val lawyerId: String = "",
    val lawyerName: String = "",
    val status: String = "", // "pending", "active", "closed", "cancelled"
    val priority: String = "", // "low", "medium", "high"
    val category: String = "",
    val createdAt: String = "",
    val updatedAt: String = "",
    val lastUpdate: String = "",
    val nextHearing: String = "",
    val nextHearingDate: String = "",
    val courtName: String = "",
    val caseNumber: String = "",
    val documents: List<Document> = emptyList()
)